NameLayerMain/src/vg/civcraft/mc/namelayer/database/GroupManagerDao.java

```diff
@@ -546,16 +546,29 @@ public void initializeStatements(){
 		
 		getAllGroupIds = db.prepareStatement("select group_id from faction_id");
 	}
+	
+	public synchronized void dbrefresh(){
+		NameLayerPlugin.reconnectAndReintializeStatements();
+	}
+	
+	public synchronized ResultSet queryStatement(PreparedStatement stat) throws SQLException{
+		return stat.executeQuery();
```

Comment:
Are you synchronizing here for any specific reason? 

This seems like a great source of deadlock/starvation if we're not careful.

If you've gone to this much effort refactoring all DB stuff into these method calls, why not add a proper connection pool?